### ISSUE TITLE
RKE1 Nodedriver Clusters - Nodes Not Being Terminated In Cloud Provider

### ISSUE THREAD

**Issue by git-ival** [rancher/rancher#35778]
**Rancher Server Setup**
- Rancher version: v2.6-head (commit: 5b56b17)
- Installation option: Helm Chart
   - Chart version: 2.6.3-rc3
   - k3s version: 1.21.4+k3s1
- Proxy/Cert Details: LetsEncrypt

**Information about the Rancher Server**
- Kubernetes version: 1.20.12+k3s1
- Cluster Type: Local

**Information about the Downstream Cluster(s)**
- Kubernetes version: 1.20.13
- Cluster Type: Downstream
   - Infrastructure Provider = AWS

**Describe the bug**
Downstream RKE1 Amazon EC2 nodedriver clusters sometimes create additional nodes in defined nodepools that are not properly removed/terminated in the cloud provider when the cluster is deleted or when nodes go into a bad state. These "rogue nodes" are also not terminated from the cloud provider after cluster deletion.

Further, if a node fails to make SSH available up to the limit of retries (60 times) then Rancher spins up additional node(s) in the nodepool. The bad and unneeded nodes are not properly terminated in the cloud provider. Once Rancher removes the nodes they are removed from the UI, but these rogue nodes are still running in the cloud provider (and seems that the pods/containers are still running on the rogue nodes as well).

**To Reproduce**
1. Create an AWS Cloud Credential
2. Create a Node Template
    1. AMI: ami-0170d35c80829aee2
    2. Use IAM Instance Profile 
    3. Docker Version: 20.10
3. Create RKE Amazon EC2 Nodedriver clusters
    1. 3 Node Pools, 1 node, all roles
    2. Network plugin: canal
    3. Addon job timeout: 60
    2. Cloud Provider: Amazon (In-Tree)
    3. Docker: Require a supported Docker version
    4. K8s version: v1.21.6-rancher1-1

**Result**

**Expected Result**
* Rancher should not attempt to provision multiple nodes in a nodepool defined to have a quantity of 1 node
* If a Node enters a bad state, fails to setup or fails to be reach-able via SSH after 60 retries (the default) then only 1 node should be provisioned to replace it

**Screenshots**
![image](https://user-images.githubusercontent.com/85186645/144917918-9d18ac98-b999-4dd2-a3d8-c7405697736a.png)
![image](https://user-images.githubusercontent.com/85186645/144918006-dcb98476-ddc4-47c5-9b9b-62bd014ffae9.png)

**Additional context**
* Tested on 2.6-head and 2.5-head, occurs with both versions
* Only tested with Amazon EC2 nodedriver
* Tested **manually and automatically** via terraform using `rancher2` and `aws` providers
* Rancher logs for a cluster experiencing this issue: [rke-provisioning-logs.zip](https://github.com/rancher/rancher/files/7663432/rke-provisioning-logs.zip)

**Comment by tunaman** [comment]
Yeah we see this as well, doesn't always happen but it's scary as hell when this happens on production

**Comment by jakefhyde** [comment]
This issue is for RKE1 downstream clusters. RKE1 is EOL as per https://support.scc.suse.com/s/kb/RKE-EOL-what-when-why. Closing the issue as "not planned".
